src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutItem.cs

```diff
@@ -73,5 +139,685 @@ internal void InvokeClick()
+			m_tpKeyboardAcceleratorTextBlock = keyboardAcceleratorTextBlock;
+
+			SuppressIsEnabled(false);
+			UpdateCanExecute();
+
+			// Sync the logical and visual states of the control
+			UpdateVisualState();
+		}
+
+		// PointerPressed event handler.
+		protected override void OnPointerPressed(PointerRoutedEventArgs pArgs)
+		{
+			base.OnPointerPressed(pArgs);
+			var handled = pArgs.Handled;
+			if (!handled)
+			{
+				PointerPoint spPointerPoint;
+				PointerPointProperties spPointerProperties;
+				spPointerPoint = pArgs.GetCurrentPoint(this);
+				spPointerProperties = spPointerPoint.Properties;
+				var bIsLeftButtonPressed = spPointerProperties.IsLeftButtonPressed;
+				if (bIsLeftButtonPressed)
+				{
+					m_bIsPointerLeftButtonDown = true;
+					m_bIsPressed = true;
+
+					pArgs.Handled = true;
+					UpdateVisualState();
+				}
+			}
+		}
+
+		// PointerReleased event handler.
+		protected override void OnPointerReleased(PointerRoutedEventArgs pArgs)
+		{
+			base.OnPointerReleased(pArgs);
+
+			bool handled = false;
+
+			handled = pArgs.Handled;
+			if (!handled)
+			{
+				m_bIsPointerLeftButtonDown = false;
+
+				m_shouldPerformActions = m_bIsPressed && !m_bIsSpaceOrEnterKeyDown && !m_bIsNavigationAcceptOrGamepadAKeyDown;
+
+#if false
+				// UNO TODO
+				if (m_shouldPerformActions)
+				{
+					GestureModes gestureFollowing = GestureModes.None;
+
+					m_bIsPressed = false;
+
+					gestureFollowing = ((PointerRoutedEventArgs*)(pArgs).GestureFollowing);
+
+					// Note that we are intentionally NOT handling the args
+					// if we do not fall through here because basically we are no_opting in that case.
+					if (gestureFollowing != GestureModes.RightTapped)
+					{
+						pArgs.Handled = true;
+						PerformPointerUpAction();
+					}
+#else
+				PerformPointerUpAction();
+#endif
+			}
+		}
+
+		 void OnRightTappedUnhandled(RightTappedRoutedEventArgs pArgs)
+		{
+			var handled = pArgs.Handled;
+			if (!handled)
+			{
+				PerformPointerUpAction();
+				pArgs.Handled = true;
+			}
+		}
+
+		// Contains the logic to be employed if we decide to handle pointer released.
+		void PerformPointerUpAction()
+		{
+			if (m_shouldPerformActions)
+			{
+				Focus(FocusState.Pointer);
+				Invoke();
+			}
+		}
+
+		// Performs appropriate actions upon a mouse/keyboard invocation of a 
+		internal void Invoke()
+		{
+			RoutedEventArgs spArgs;
+			MenuFlyoutPresenter spParentMenuFlyoutPresenter;
+
+			// Create the args
+			spArgs = new RoutedEventArgs();
+			spArgs.OriginalSource = this;
+
+			// Raise the event
+			Click?.Invoke(this, spArgs);
+
+			// UNO TODO
+			//AutomationPeer.RaiseEventIfListener(this, xaml_automation_peers.AutomationEvents_InvokePatternOnInvoked);
+
+			// Execute command associated with the button
+			ExecuteCommand();
+
+			bool shouldPreventDismissOnPointer = false; // UNO TODO PreventDismissOnPointer;
+			if (!shouldPreventDismissOnPointer)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)